repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolProviderTest.groovy

```diff
@@ -107,8 +125,124 @@ class HttpConnectionPoolProviderTest {
     public void "should not add header parameter when not configured"() {
         poolType.setHeaders(null)
 
-        DefaultHttpClient client = HttpConnectionPoolProvider.genClient(poolType) as DefaultHttpClient
+        DefaultHttpClient client = HttpConnectionPoolProvider.genClient("", poolType) as DefaultHttpClient
 
         assert !client.getParams().getParameter(ClientPNames.DEFAULT_HEADERS)
     }
+
+    @Test
+    public void "should create a pool that can connect to a server with client auth using keystore and truststore when configured"() {
+        server = new Server()
+
+        // SSL Context Factory
+        def sslContextFactory = new SslContextFactory()
+        sslContextFactory.keyStoreResource = Resource.newClassPathResource("server.jks")
+        sslContextFactory.keyStorePassword = "password"
+        sslContextFactory.keyManagerPassword = "password"
+        sslContextFactory.needClientAuth = true
+        sslContextFactory.trustStoreResource = Resource.newClassPathResource("client.jks")
+        sslContextFactory.trustStorePassword = "password"
+
+        // SSL HTTP Configuration
+        def https_config = new HttpConfiguration()
+        https_config.addCustomizer new SecureRequestCustomizer()
+
+        // SSL Connector
+        def sslConnector = new ServerConnector(
+                server,
+                new SslConnectionFactory(sslContextFactory, HttpVersion.HTTP_1_1.asString()),
+                new HttpConnectionFactory(https_config)
+        )
+        sslConnector.setPort(0)
+
+        // Start the server
+        def statusCode = HttpServletResponse.SC_OK
+        def responseContent = "The is the plain text test body data.\n".bytes
+        def contentType = "text/plain;charset=utf-8"
+        // Make this the only endpoint for the server
+        server.connectors = [sslConnector] as Connector[]
+        server.handler = new AbstractHandler() {
+            @Override
+            void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+                response.status = statusCode
+                response.contentType = contentType
+                baseRequest.handled = true
+                response.outputStream.write responseContent
+            }
+        }
+        server.start()
+        def serverPort = ((ServerConnector) server.connectors[0]).getLocalPort()
+
+        // Add the client creds to the connection pool
+        poolType.setKeystoreFilename(HttpConnectionPoolProviderTest.class.getResource("/client.jks").getFile())
+        poolType.setKeystorePassword("password")
+        poolType.setKeyPassword("password")
+        poolType.setTruststoreFilename(HttpConnectionPoolProviderTest.class.getResource("/server.jks").getFile())
+        poolType.setTruststorePassword("password")
+
+        def configRoot = new File(HttpConnectionPoolProviderTest.class.getResource("/client.jks").getFile()).getParent()
```

Comment:
I've seen this getResource call a bunch. Not that it makes a difference in the end, but it may be easier to read to just get these file locations up-front and bind them to some well-named variable.